### PR TITLE
Galaxy 1692 ui tests for tab config

### PR DESCRIPTION
**Pull request by emmaaa**
Two little espresso tests, our starting point for getting things running on CI 🌈 

The bulk of this PR is just getting the structure in place so that we can add tests more easily in future
If we need to change the tests themselves in future, that's not a big deal :) 
image

### CODE REVIEW

**Comment by emmaaa** [comment]
image

File: app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/matchers/RecyclerViewMatchers.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                resources!!.getResourceName(recyclerViewId)
+            } catch (var4: Resources.NotFoundException) {
+                String.format("%s (resource name not found)", recyclerViewId)
+            }
+
+            description.appendText("RecyclerView with id: $idDescription at position: $position")
+        }
+
+        public override fun matchesSafely(view: View): Boolean {
+            resources = view.resources
```
Comment (TomMarshBBC) [inline app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/matchers/RecyclerViewMatchers.kt]: is it somehow possible to put this into a `when` statement, or split this out? it's a bit hard to read

File: app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/assertions/ViewAssertions.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        onView(viewMatcher).assertIsDisplayed()
+    }
+
+    fun assertDisplayingViews(vararg viewIds: Int) {
+        for (viewId in viewIds)
+            onView(withId(viewId)).assertIsDisplayed()
+    }
+
+    fun assertDisplayingViews(vararg viewMatchers: Matcher<View>) {
+        for (viewMatcher in viewMatchers)
```
Comment (TomMarshBBC) [inline app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/assertions/ViewAssertions.kt]: for all these, `for` statements, I personally prefer something like  (but up to you)

```
    viewMatchers.forEach { 
            onView(it).assertIsDisplayed()
        }
```

File: build.gradle [hunk 3]
```diff
@@ -76,7 +76,7 @@ allprojects {
         androidx_version = '1.1.0'
         androidx_appcompat_version = '1.1.0-beta01'
         androidx_annotation_version = '1.0.2'
-        androidx_test_runner_rules_version = '1.2.0'
```
[the remark this answers is not in the document]
Comment (emmaaa) [reply build.gradle]: when it's set to 1.2.0 it does this
image

File: app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/matchers/RecyclerViewMatchers.kt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    fun atPosition(position: Int): Matcher<View> = atPositionOnView(position, -1)
+
+    private fun atPositionOnView(position: Int, targetViewId: Int): Matcher<View> = object : TypeSafeMatcher<View>() {
+        var resources: Resources? = null
+        var childView: View? = null
+
+        override fun describeTo(description: Description) {
+            var idDescription = recyclerViewId.toString()
+            if (resources != null) idDescription = try {
+                resources!!.getResourceName(recyclerViewId)
```
[the remark this answers is not in the document]
Comment (emmaaa) [reply app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/matchers/RecyclerViewMatchers.kt]: awesome! Yeah this was originally Java so it was a quick convert without much fancy kotlinising. I'll have a go with that 👍

File: app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/assertions/ViewAssertions.kt (hunk 2 above)
Comment (emmaaa) [reply app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/assertions/ViewAssertions.kt]: i never thought of that 👀 yeah that looks much neater

File: app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/arrangements/UserArrangements.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Rule
+    private val activityRule = ActivityTestRule(MainActivity::class.java, true, false)
+    @Rule
+    private val ribbonActivityRule = ActivityTestRule(RibbonActivity::class.java, true, false)
+    @Rule
+    private val bottomNavActivityRule = ActivityTestRule(BottomNavigationActivity::class.java, true, false)
+
+    fun launchesTheAppWithRibbonEnabled() {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
```
[the remark this answers is not in the document]
Comment (johnryM) [reply app/src/androidTest/java/bbc/co/uk/nautilusapp/ui/arrangements/UserArrangements.kt]: I did this for for Rubik in the `ExpressoHelper` object so might be good to add that here.

File: app/build.gradle [hunk 6]
```diff
@@ -27,6 +27,9 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+        debug {
```
Comment (TomMarshBBC) [inline app/build.gradle]: remove this debug, we actually don't need it

Comment (TomMarshBBC) [reply app/build.gradle]: as in the whole block

**Pull request merged**